src/java.desktop/share/classes/sun/java2d/pipe/RenderingEngine.java

```diff
@@ -218,6 +222,47 @@ public abstract void strokeTo(Shape src,
                                   boolean antialias,
                                   PathConsumer2D consumer);
 
+    /**
+     * Sends the geometry for a widened path as specified by the parameters
+     * to the specified consumer.
+     * <p>
+     * The specified {@code src} {@link Shape} is widened according
+     * to the parameters specified by the {@link BasicStroke} object.
+     * Adjustments are made to the path as appropriate for the
+     * {@link java.awt.RenderingHints#VALUE_STROKE_NORMALIZE} hint if the
+     * {@code normalize} boolean parameter is true.
+     * Adjustments are made to the path as appropriate for the
+     * {@link java.awt.RenderingHints#VALUE_ANTIALIAS_ON} hint if the
+     * {@code antialias} boolean parameter is true.
+     * <p>
+     * The geometry of the widened path is forwarded to the indicated
+     * {@link PathConsumer2D} object as it is calculated.
+     *
+     * @param src the source path to be widened
+     * @param at the transform to be applied to the shape and the
+     *           stroke attributes
+     * @param clip the current clip in effect in device coordinates
+     * @param bs the {@code BasicSroke} object specifying the
+     *           decorations to be applied to the widened path
+     * @param thin true if the transformed stroke attributes are smaller
+     *             than the minimum dropout pen width
+     * @param normalize indicates whether stroke normalization should
+     *                  be applied
+     * @param antialias indicates whether or not adjustments appropriate
+     *                  to antialiased rendering should be applied
+     * @param consumer the {@code PathConsumer2D} instance to forward
+     *                 the widened geometry to
+     * @since 17
+     */
+    public abstract void strokeTo(Shape src,
```

Comment:
I think we need to clarify the backward compatibility of this change.
 - If we would like to support the old marlin version via "-Dsun.java2d.renderer" option in jdk17 then adding the new abstract method to this interface breaks this possibility. To support that this method should be made default and call the old strokeTo method.
 - If support for the old marlin library is not necessary then we could drop the old method from this interface(but preserve it in the MarlinRenderingEngine to have the same marlin codebase), it won't be used anyway.

Reply:
I like the first proposal: use default class method as RE is an abstract class.